WPFWijkagent/WPFWijkagent/WitnessController.cs

```diff
@@ -0,0 +1,101 @@
+                $"\nTweet of retweet met de hashtag #Delict{_Offence.ID} als u meer weet.");
+            }
+            
+            else
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Checks if Message already existst in the database
+        /// </summary>
+        /// <returns>true is the message exists, false if message does not exist</returns>
+        public bool MessageExists()
+        {
+            DBContext dBContext = new DBContext();
+            SqlCommand query = new SqlCommand("SELECT TOP(1) OffenceID FROM SendMessage WHERE Message = @Message");
+            query.Parameters.Add("@Message", System.Data.SqlDbType.NVarChar);
+            query.Parameters["@Message"].Value = CreateWitnessMessage();
+            List<object[]> rows = dBContext.ExecuteSelectQuery(query);
+
+            if(rows.Count > 0)
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// connect to the twitter bot and publish the custom tweet.
+        /// Insert the created hashtag in the database.
+        /// save the created message to avoid duplicates
+        /// </summary>
+        public void SendTweet(MainWindow window)
+        {
+            DBContext dBContext = new DBContext();
+            Scraper scraper = new Scraper(_Offence);
+            scraper.Connect();
+
+            Tweet.PublishTweet(CreateWitnessMessage());
+            SqlCommand query = new SqlCommand("UPDATE Offence SET Hashtag = @Hashtag WHERE ID = @OffenceID");
+
+            //prepare values in statement 
+            query.Parameters.Add("@OffenceID", System.Data.SqlDbType.Int);
+            query.Parameters.Add("@Hashtag", System.Data.SqlDbType.NVarChar);
+
+            query.Parameters["@Hashtag"].Value = $"Delict{_Offence.ID}";
+            query.Parameters["@OffenceID"].Value = _Offence.ID;
+
+            SqlCommand insertMessage = new SqlCommand("INSERT INTO SendMessage (OffenceID, Message) OUTPUT INSERTED.ID" +
+                " VALUES (@OffenceID, @Message)");
+            insertMessage.Parameters.Add("@OffenceID", System.Data.SqlDbType.Int);
+            insertMessage.Parameters.Add("@Message", System.Data.SqlDbType.NVarChar);
+
+            insertMessage.Parameters["@OffenceID"].Value = _Offence.ID;
+            insertMessage.Parameters["@Message"].Value = CreateWitnessMessage();
+
+            dBContext.ExecuteQuery(query);
+            dBContext.ExecuteInsertQuery(insertMessage);
+            _Offence.CallHashtag = $"#Delict{_Offence.ID}";
+            Scraper WitnessScraper = new Scraper(_Offence, true, window.Hashtag(_Offence)); ;
+            WitnessScraper.UpdateSocialMediaMessages(1);
```

Comment:
een aantal dingen hier:
we werken in oop dus alle functionaliteit is onderschijden van elkaar. jij doet hier 3 dingen:

1. tweet publishen en verversen van tweets (die tweede scraper had hierbij waarschijnlijk niet gehoeven als je de search parameters na initializatie nog kon aanpassen.)
2. updaten van offence (dit moet in de offence controller)
3. toevoegen van sendmessage(hiervoor moet een apparte controller "sendmessagecontroller" of een methode bij social media message alleen klopt dit niet met onze architectuur....)